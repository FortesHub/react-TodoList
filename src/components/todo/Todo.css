.task {
  padding: 15px;
  margin: 14px;
  list-style: none;
  border: 2px solid;
  border-radius: 12px;
  border-style: inset;
  border-color: black;
  cursor: pointer;
  width: 90%;

}

.task span {
  color: #000000;
  font-weight: bold;
}

.editTodo{
  background-color: bisque ;
  border-style: dotted ;
  padding: 5px;
  width: 85%;
  font-size: 16px;
  color: sienna
}

.icon {
  background-color: bisque;
  border: none;
  color: black;
  float: right;
  margin-right: 5px;
}
