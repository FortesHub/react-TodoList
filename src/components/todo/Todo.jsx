import React from "react";
import { useDispatch } from "react-redux";
import { useState } from "react";

import "./Todo.css";
import { toggleTodo, updateTodo, deleteTodo } from "../../redux/actions";

const Todo = ({ todo }) => {
  const [editing, setEditing] = useState(false);
  const [text, setText] = useState(todo?.data);
  const onFormSubmit = (e) => {
    e.preventDefault();
    setEditing((prevState) => !prevState);
    dispatch(updateTodo(todo?._id, text));
  };
  const dispatch = useDispatch();
  return (
    <li
      className="task"
      onClick={() => dispatch(toggleTodo(todo?._id))}
      style={{
        textDecoration: todo?.done ? "line-through 4px" : "",
        color: todo?.done ? "#ff0000" : "#000000",
      }}
      data-testid="todo-test"
    >
      <span style={{ display: editing ? "none" : "" }}>{todo?.data}</span>
      <form
        style={{ display: editing ? "inline" : "none" }}
        onSubmit={onFormSubmit}
      >
        <input
          className="editTodo"
          type="text"
          value={text}
          onChange={(e) => setText(e.target.value)}
        />
      </form>
      <button 
      className="icon"
       title="Delete"
       onClick={() => dispatch(deleteTodo(todo?._id))}
       >
        <i className="fa fa-trash"></i>
      </button>
      <button
        className="icon"
        title="Edit"
        onClick={() => setEditing((prevState) => !prevState)}
      >
        <i className="fa fa-thin fa-pen-to-square"></i>
      </button>
    </li>
  );
};

export default Todo;
