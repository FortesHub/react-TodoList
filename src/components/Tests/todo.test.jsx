import React from "react";
import { render, screen } from '@testing-library/react';
import configStore from 'redux-mock-store';
import Todo from '../todo/Todo';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';


test('should render our component', () => {
    const initialState = { state: []};
    const mockStore = configStore();

    const store = mockStore(initialState);

    render(
        <Provider store={store}>
        <Todo/>
        </Provider>
    );

    const todoElement = screen.getByTestId('todo-test');
    expect(todoElement).toBeInTheDocument();

})

test('should render our todos', () => {
    const initialState = { state: []};
    const mockStore = configStore();

    const store = mockStore(initialState);

    const todo = { data: 'This is a Task', done: true, createdAt: ''}
    render(
        <Provider store={store}>
        <Todo todo={todo}/>
        </Provider>
    );

    const todoElement = screen.getByTestId('todo-test');
    expect(todoElement).toHaveTextContent('This is a Task');
    expect(todoElement).toContainHTML('</form>');
    expect(todoElement).not.toContainHTML('<ul>');
})

test('match snapshot', () => {
    const initialState = { state: []};
    const mockStore = configStore();

    const store = mockStore(initialState);

    const todo = { data: 'This is a Task', done: true, createdAt: ''}

  const tree = renderer.create(
        <Provider store={store}>
        <Todo todo={todo} />
        </Provider>
    ).toJSON();

    console.log(tree);

    expect(tree).toMatchSnapshot();

})