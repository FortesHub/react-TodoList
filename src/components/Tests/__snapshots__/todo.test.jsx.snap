// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<li
  className="task"
  data-testid="todo-test"
  onClick={[Function]}
  style={
    Object {
      "color": "#ff0000",
      "textDecoration": "line-through 4px",
    }
  }
>
  <span
    style={
      Object {
        "display": "",
      }
    }
  >
    This is a Task
  </span>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <input
      className="editTodo"
      onChange={[Function]}
      type="text"
      value="This is a Task"
    />
  </form>
  <button
    className="icon"
    onClick={[Function]}
    title="Delete"
  >
    <i
      className="fa fa-trash"
    />
  </button>
  <button
    className="icon"
    onClick={[Function]}
    title="Edit"
  >
    <i
      className="fa fa-thin fa-pen-to-square"
    />
  </button>
</li>
`;
